src/frontend/verify-jira-jwt-middleware.ts

```diff
@@ -7,10 +7,11 @@ const verifyJiraJwtMiddleware = (tokenType: TokenType) => async (
 	res: Response,
 	next: NextFunction
 ): Promise<void> => {
-	const jiraHost = (req.query.xdm_e as string) || req.body?.jiraHost;
+	const jiraHost = req.body?.jiraHost || req.session.jiraHost;
```

Comment:
Why can't we use `req.query.xdm_e` here? If it doesn't match JWT token there will be 401. So should be secure to keep it. (Also, wouldn't it break the Jira iframe page?)

Reply:
We can, but that means scattering the logic of `xdm_e` and `jiraHost` across multiple files.  I find it simpler to have it in a single location.  It doesn't break it because we set it in the session from the middleware in app.ts, where the rest of the logic resides.  

That being said, I've been playing with the idea that we should be using `res.locals.jiraHost` instead so we can truly 'centralize' the logic into a single variable and then we just need a single middleware that can do all the logic.

Reply:
I've improved this by using `res.locals.jiraHost` as the main entrypoint throughout the codebase for jiraHost.  Now we only have a single location that does all the logic for everything, let it be `xdm_e` query, included in body or from session.